exporter/googlecloudexporter/internal/resourcemapping/custommapping_test.go

```diff
@@ -0,0 +1,79 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package resourcemapping
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	monitoredrespb "google.golang.org/genproto/googleapis/api/monitoredres"
+)
+
+func TestCustomMonitoredResourceMapping(t *testing.T) {
+	tests := []struct {
+		name     string
+		resource pcommon.Resource
+		want     *monitoredrespb.MonitoredResource
+	}{
+		{
+			name: "Custom Monitored Resource",
+			resource: func() pcommon.Resource {
+				r := pcommon.NewResource()
+				r.Attributes().PutStr("gcp.resource_type", "custom_resource")
+				r.Attributes().PutStr("gcp.custom_resource.label1", "value1")
```

Comment:
There is a chance of future collision with MR types and the GCP resource conventions https://github.com/open-telemetry/semantic-conventions/tree/main/docs/resource/cloud-provider/gcp

Maybe a non issue but wanted to call it out

Reply:
The collision should not happen for custom resources since they will not be a part of the semconv. However care must be taken that strings used to represent custom MR are different from those defined in semconv.